Sources/AccioKit/Services/CachedBuilderService.swift

```diff
@@ -12,7 +12,8 @@ final class CachedBuilderService {
     func frameworkProducts(manifest: Manifest, appTarget: AppTarget, dependencyGraph: DependencyGraph, platform: Platform) throws -> [FrameworkProduct] {
         var frameworkProducts: [FrameworkProduct] = []
 
-        for framework in try appTarget.frameworkDependencies(manifest: manifest, dependencyGraph: dependencyGraph).flattenedDeepFirstOrder() {
+        let frameworks = Set(try appTarget.frameworkDependencies(manifest: manifest, dependencyGraph: dependencyGraph).flattenedDeepFirstOrder())
```

Comment:
Hmm, I'm not sure if this is the approach we should take since this changes the order of how the frameworks will be compiled which could be a problem if they are dependent on each other. I currently don't have to time to dig into this and ensure this is correct, but in any case it would be safer to just implement a `removeDuplicates` method which keeps sorting stable. It _could_ use a set as an implementation detail, but the end result should be a sorted list, so an array, IMHO.

Reply:
Another approach that may be worth considering is to avoid adding the frameworks if they are already in the array. I think that can be done [here](https://github.com/JamitLabs/Accio/blob/stable/Sources/AccioKit/Models/AppTarget.swift#L43)

Reply:
@acecilia I guess it isn't possible to fix this at the place you suggested, as the `.flattenedDeepFirstOrder` call (where the required frameworks come in) is only called after the `frameworkDependencies` call